r/R/matrix.R

```diff
@@ -158,6 +157,36 @@ all_matrix_inputs <- function(x) {
   x
 }
 
+#' Get the max of each row in an iterable matrix
+#' @param x IterableMatrix object/dgCMatrix object
+#' @export
+row_max <- function(x) {
+  iter <- iterate_matrix(convert_matrix_type(x, "double"))
+  if(x@transpose == TRUE) {
+    res <- matrix_compute_max_per_col(iter)
+  } else {
+    res <- matrix_compute_max_per_row(iter)
+  }
+  names(res) <- rownames(x)
+  res
+}
+
+
+#' Get the max of each col in an interable matrix
+#' @param x IterableMatrix/dgCMatrix object
+#' @export
```

Comment:
One other nice trick we can do for the docs is using `@describeIn` or `@rdname` to group functions together into a single documentation page. For example, see [open_fragments_10x](https://bnprks.github.io/BPCells/reference/open_fragments_10x.html) docs and implementation.

For full docs see [roxygen2](https://roxygen2.r-lib.org/articles/reuse.html)